File: easy/0559_maximum_depth_of_n_ary_tree/Solution.java
import nodes.Node;

class Solution {
  public int maxDepth(Node root) {
    if (root == null) {
      return 0;
    }

    int maxDepth = 0;

    for (final Node child : root.children) {
      maxDepth = Math.max(maxDepth, maxDepth(child));
    }

    return 1 + maxDepth;
  }
}

File: easy/0559_maximum_depth_of_n_ary_tree/solution.py
from utils.python.nodes import Node


class Solution:
    def maxDepth(self, root: Node) -> int:
        stack = [(root, 1)]

        max_depth = 1

        while stack:
            node, depth = stack.pop()
            max_depth = max(max_depth, depth)

            for child in node.children:
                stack.append((child, depth + 1))

        return max_depth
